manifests/k3os-operator.yaml

```diff
@@ -0,0 +1,165 @@
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: k3os-system
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  labels:
+    apps.kubernetes.io/component: controller
+    apps.kubernetes.io/name: k3os-operator
+  name: k3os-operator
+  namespace: k3os-system
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  labels:
+    apps.kubernetes.io/component: controller
+    apps.kubernetes.io/name: k3os-operator
+  name:  k3os-operator
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: cluster-admin
+subjects:
+- kind: ServiceAccount
+  name: k3os-operator
+  namespace: k3os-system
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: updatechannels.k3os.cattle.io
+  namespace: k3os-system
+spec:
+  group: k3os.cattle.io
+  names:
+    kind: UpdateChannel
+    plural: updatechannels
+    singular: updatechannel
+    shortNames:
+    - upchan
+    categories:
+    - all
+  scope: Namespaced
+  versions:
+  - name: v1
+    served: true
+    storage: true
+    schema:
+      openAPIV3Schema:
+        type: object
+        properties:
+          apiVersion:
+            type: string
+          kind:
+            type: string
+          spec:
+            type: object
+            properties:
+              concurrency:
+                type: integer
+              version:
+                type: string
+              url:
+                type: string
+          status:
+            type: object
+            properties:
+              polling:
+                type: string
+              upgrading:
+                type: array
+                items:
+                  type: string
+    # subresources:
+    #   status: {}
```

Comment:
I've tried the subresource route but updates resulted in empty objects. I think some wrangler magic is missing.